core/src/main/scala/org/apache/spark/deploy/rest/StandaloneRestClient.scala

```diff
@@ -289,6 +337,14 @@ private[deploy] class StandaloneRestClient extends Logging {
   private def handleUnexpectedRestResponse(unexpected: SubmitRestProtocolResponse): Unit = {
     logError(s"Error: Server responded with message of unexpected type ${unexpected.messageType}.")
   }
+
+  private def handleSubmitRestConnectionException(url: String): Boolean = {
```

Comment:
I would just call this `handleConnectionException` and rename the parameter to `masterUrl`. It would also be good to add a java doc here.